File: backend/src/TaskMgmt.Domain/Entities/Attachment.cs
using TaskMgmt.Domain.Common;

namespace TaskMgmt.Domain.Entities;

public class Attachment : AuditableEntity
{
    public required Guid WorkTaskId { get; set; }
    public WorkTask? WorkTask { get; set; }

    // Tên file gốc do người dùng upload, chỉ dùng để hiển thị/tải xuống.
    public required string FileName { get; set; }

    // Khoá lưu trữ ngẫu nhiên trong bucket (không dựa vào FileName) để tránh path traversal/đè file.
    public required string StorageKey { get; set; }

    public required string ContentType { get; set; }
    public required long SizeBytes { get; set; }

    // Ánh xạ tới CreatedByUserId (kế thừa từ AuditableEntity) để hiển thị người upload.
    public User? Uploader { get; set; }
}

File: mobile/taskmgmt_app/lib/features/attachments/domain/entities/attachment.dart
class Attachment {
  const Attachment({
    required this.id,
    required this.workTaskId,
    required this.fileName,
    required this.contentType,
    required this.sizeBytes,
    required this.uploadedByUserId,
    required this.uploadedByFullName,
    required this.uploadedByEmail,
    required this.createdAtUtc,
  });

  final String id;
  final String workTaskId;
  final String fileName;
  final String contentType;
  final int sizeBytes;
  final String? uploadedByUserId;
  final String uploadedByFullName;
  final String uploadedByEmail;
  final DateTime createdAtUtc;

  bool get isImage => contentType.startsWith('image/');
}
